app/components/ui/Navigation/MobileNavigation.tsx

```diff
@@ -19,24 +20,41 @@ export const MobileNavigation = ({ menuData }: MobileNavigationProps) => {
   ): menuItem is NavigationMenu {
     return "link" in menuItem;
   }
+
+  const filterMenuRef = useRef<HTMLDivElement>(null);
+
   const [activeMobileSubMenu, setActiveMobileSubMenu] = useState("");
   const closeMobileMenu = () => {
     setActiveMobileSubMenu("");
+    setMobileNavigationIsOpen(false);
+  };
+  const goBackToTopMenu = () => {
+    setActiveMobileSubMenu("");
   };
   const [mobileNavigationIsOpen, setMobileNavigationIsOpen] = useState(false);
-  const toggleMobileNav = () => setMobileNavigationIsOpen((prev) => !prev);
   const mobileSubMenuIsActive = !!activeMobileSubMenu;
-  const inSubNavigation = mobileNavigationIsOpen && mobileSubMenuIsActive;
   const mobileNavIconDisplay = mobileNavigationIsOpen ? "xmark" : "bars";
   const mobileNavTextDisplay = mobileNavigationIsOpen ? "Close" : "Menu";
 
+  useClickOutside(
+    filterMenuRef,
+    (event: MouseEvent) => {
+      // Prevents collison between handling the outside click and clicking on the
+      // "Close" button
+      if ((event.target as HTMLElement).innerText !== "Close") {
+        setMobileNavigationIsOpen(false);
+      }
+    },
+    mobileNavigationIsOpen
+  );
+
   const handleOpenMobileNavigation = () => {
-    if (inSubNavigation) {
+    if (mobileSubMenuIsActive) {
       setActiveMobileSubMenu("");
-    } else if (mobileSubMenuIsActive) {
-      setMobileNavigationIsOpen(true);
+    } else if (mobileNavigationIsOpen) {
+      setMobileNavigationIsOpen(false);
     } else {
-      toggleMobileNav();
+      setMobileNavigationIsOpen(true);
```

Comment:
**note:** This logic wasn't quite exact enough considering new dynamics to click outside the menu and close.